pkg/doc/did/doc.go

```diff
@@ -0,0 +1,202 @@
+/*
```

Comment:
@llorllale @fqutishat @rolsonquadras Any thoughts on the directory structure?

this is now pkg/doc/did.Doc

Reply:
@troyronda I'm not sure of the need for that top-level `doc`, but I care far more about referencing `did.Doc` as the type all throughout client code, so I'm generally OK with this

Reply:
@troyronda I agree with @llorllale's comments